src/wasm.rs

```diff
@@ -207,18 +136,21 @@ impl Lsp {
     /// `await`ed. However, as it returns a Promise, it's a good idea to attach
     /// handlers for completion and error. If the promise ever resolves, the server
     /// is no longer running, which may serve as a hint that attention is needed.
-    pub fn run(self) -> js_sys::Promise {
-        let (service, messages) =
-            lspower::LspService::new(|_client| LspServer::default());
-        let server = lspower::Server::new(
-            Incoming {
-                messages: self.incoming.clone(),
-            },
-            Outgoing::new(Arc::new(self)),
-        )
-        .interleave(messages);
-        let future =
-            std::panic::AssertUnwindSafe(server.serve(service));
+    pub fn run(mut self) -> js_sys::Promise {
```

Comment:
A problem I bumped into while trying to integrate the `0.7.0` flux-lsp into vscode: `server.run` _consumes_ `server` so it can no longer be used to `server.send` anymore. The way we test, we always call `server.run` last so we never see the error. In practice, that's kind of a show-stopper bug.

Reply:
I got around that by having an internal handler for the messages that I `mem::take`. This means the `run` method can own the important bits while the server is still running. It also means that registering any `onMessage` handlers must occur before running the server. I added a comment to that effect for this change.